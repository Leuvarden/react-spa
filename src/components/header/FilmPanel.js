import React, {Component} from 'react';

export default class FilmPanel extends Component {
    render () {
        return (
            <section className="film-panel">
                <div>
                    <span className="logo">netflixroulette</span>
                    <button className="film-panel__button-search">Search</button>
                </div>
                <figure>
                    <div className="film-panel_poster"> 
                        <img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQIbwrzbJdCnh06TnSF36p7gxWAlDKfSNiChu3HyZ1ccSsHW_Q7XQ"></img>
                    </div>
                    <figcaption className="film-panel_description">
                        <h1>Pulp Fiction</h1>
                        <p className="film-panel_category">Oscar-winning movies</p>
                        <p>
                            <span className="film-panel_year">1994</span>
                            <span className="film-panel_duration">154 min</span>
                        </p>
                        <p className="film-panel_overview"></p>  
                    </figcaption>
                </figure>
            </section>
        );
    }
}