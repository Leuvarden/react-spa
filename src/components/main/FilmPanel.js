import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class FilmPanel extends Component {
    render () {
        const props = this.props;

        return (
            <section className="film-panel">
                <figure>
                    <div className="film-panel_poster"> 
                        <img src={props.img}></img>
                    </div>
                    <figcaption className="film-panel_description">
                        <h1>{props.title}</h1>
                        <p className="film-panel_category">{props.genres[0]}</p>
                        <p>
                            <span className="film-panel_year">{props.date.slice(0, 4)}</span>
                            {/* <span className="film-panel_duration">154 min</span> */}
                        </p>
                        <p className="film-panel_overview">{props.overview}</p>  
                    </figcaption>
                </figure>
            </section>
        );
    }
}

FilmPanel.propTypes = {
    genres: PropTypes.array,
    overview: PropTypes.string,
    title: PropTypes.string,
    img: PropTypes.string,
    date: PropTypes.string
};