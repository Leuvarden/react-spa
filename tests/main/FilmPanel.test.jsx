import React from 'react';
import { FilmPanel } from './../../src/components/main/FilmPanel.js';

/* eslint-disable no-undef */
describe ('FilmPanel', () => {

    const defaultProps = {
        genres: [ 'Adventure', 'Science'],
        poster_path: 'poster_path',
        release_date: '2014-05-02',
        title:'Pulp Fiction',
        overview:'Lorem ipsum dolem sit amet'
    };

    it('renders as expected', () => {
        const item = shallow(
            <FilmPanel {...defaultProps}/>
        );
    
        expect(item).toMatchSnapshot();
    });

});