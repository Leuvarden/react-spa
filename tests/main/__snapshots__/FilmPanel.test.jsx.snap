// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPanel renders as expected 1`] = `
<section
  className="film-panel"
>
  <figure>
    <div
      className="film-panel_poster"
    >
      <img
        src="poster_path"
      />
    </div>
    <figcaption
      className="film-panel_description"
    >
      <h1>
        Pulp Fiction
      </h1>
      <p
        className="film-panel_category"
      >
        Adventure
      </p>
      <p
        className="film-panel_year"
      >
        2014
      </p>
      <p
        className="film-panel_overview"
      >
        Lorem ipsum dolem sit amet
      </p>
    </figcaption>
  </figure>
</section>
`;
